.github/workflows/macos_ui_tests.yml

```diff
@@ -317,8 +353,26 @@ jobs:
         path: "macOS/${{ steps.prepare-test-report.outputs.file }}"
         retention-days: 1
 
+    - name: Upload xcresult files
+      uses: actions/upload-artifact@v4
+      if: always() && steps.check-failures.outputs.had_failures == 'true'
+      with:
+        name: "xcresult-${{ matrix.test }}-macos${{ matrix.os-version }}"
+        path: "macOS/DerivedData/Logs/Test/*.xcresult"
+        retention-days: 7
```

Comment:
I’d consider lowering this to 1 day, otherwise our storage usage may grow significantly when there are multiple failures.